Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/DXResourceBinding.cpp

```diff
@@ -0,0 +1,59 @@
+#define NOMINMAX
+#define WIN32_LEAN_AND_MEAN
+
+#include "pch.h"
+#include "DXResourceBinding.h"
+#include "DXResourceBinding.g.cpp"
+#include "stdafx.h"
+#include "ORTHelpers.h"
+
+using namespace winrt::Microsoft::AI::MachineLearning;
+
+#undef min
+
+Microsoft::WRL::ComPtr<ID3D12Resource> d3dResource;
+static std::optional<Ort::Session> preprocesingSession;
+static std::optional<Ort::Session> inferenceSession;
+D3D12Quad sample(800, 600, L"D3D12 Quad");
+
+namespace winrt::WinMLSamplesGalleryNative::implementation
+{
+    // Create ORT Sessions and launch D3D window in a separate thread
+	void DXResourceBinding::LaunchWindow() {
+        // Create ORT Sessions that will be used for preprocessing and classification
+        preprocesingSession = CreateSession(Win32Application::GetAssetPath(L"dx_preprocessor_efficient_net.onnx").c_str());
+        inferenceSession = CreateSession(Win32Application::GetAssetPath(L"efficientnet-lite4-11.onnx").c_str());
+
+        // Spawn the window in a separate thread
+        std::thread d3d_th(Win32Application::Run, &sample, 10);
+
+        // Wait until the D3D pipeline finishes
+        while (!sample.is_initialized) {}
+
+        // Detach the thread so it doesn't block the UI
+        d3d_th.detach();
+	}
+
+    // Get the buffer currently being drawn to the screen then
+    // preprocess and classify it
+    winrt::com_array<float> DXResourceBinding::EvalORT() {
+        // Get the buffer currently being drawn to the screen
+        ComPtr<ID3D12Resource> currentBuffer = sample.GetCurrentBuffer();
```

Comment:
while running this locally, "currentBuffer" was null and crashed when passing that value into ORT. Looks like this code should handle that case.

Reply:
Done.